llvm/lib/CodeGen/SjLjEHPrepare.cpp

```diff
@@ -442,9 +442,19 @@ bool SjLjEHPrepare::setupEntryBlockAndCallSites(Function &F) {
   for (BasicBlock &BB : F) {
     if (&BB == &F.front())
       continue;
-    for (Instruction &I : BB)
-      if (I.mayThrow())
-        insertCallSiteStore(&I, -1);
+    for (Instruction &I : BB) {
+      // Partially revert b61fd7f modifications.  Stop using "I.mayThrow()"
+      // here.  Using it inserts no-action marker to just before invoke
+      // instructions like "invoke void @__cxa_bad_cast()".  That means
+      // tests having exception and try/catch in the identical function
+      // are broken.  For example, tests like test_aux_runtime.pass.cpp.
+      if (CallInst *CI = dyn_cast<CallInst>(&I)) {
```

Comment:
Should this be CallBase

Reply:
Thank you for the correction.

Reply:
Actually, it's not.  We want to check only CallInst.  We don't want to check InvokeInst here.